torch/csrc/jit/script/compiler.cpp

```diff
@@ -1616,6 +1625,16 @@ struct to_ir {
         << "indexing on a non-tensor type.";
     }
     size_t dim = 0;
+
+    // Accumulate tensor Values here; call aten::index on them later.
+    std::vector<Value*> tensor_indices;
+
+    auto handle_tensor = [&](Value* tensor) {
+      tensor_indices.resize(dim + 1);
+      tensor_indices[dim] = tensor;
+      dim++;
```

Comment:
How is that different than `tensor_indices.push_back(tensor); dim++`? If the vector can have holes can you add a comment saying how exactly those indices should be understood? Do they correspond to indexing before or after processing the slices?

Reply:
Also, it seems to be used in a single place only, so it shouldn't be necessary.

Reply:
Yes, the vector can have holes in it. I'll add a comment saying how those are understood (`at::index` requires holes at indexes that aren't being indexed). 

handle_tensor is used in one place but in the future it'll be used in more places. Supporting indexing with sequences can be implemented by converting the sequence to a tensor and then using `at::index` on it.